mod-dcb/src/main/resources/volaris/mod-dcb/features/cancelling-lending-flow.feature

```diff
@@ -0,0 +1,208 @@
+    When method PUT
+    Then status 200
+
+    Given path 'transactions' , transactionId , 'status'
+    When method GET
+    Then status 200
+    And match $.status == 'AWAITING_PICKUP'
+    And match $.role == 'LENDER'
+
+    Given path 'transactions' , transactionId , 'status'
+    And request updateToCancelRequest
+    When method PUT
+    Then status 200
+
+    Given path 'transactions' , transactionId , 'status'
+    When method GET
+    Then status 200
+    And match $.status == 'CANCELLED'
+    And match $.role == 'LENDER'
+
+    Given path 'request-storage', 'requests'
+    Given param query = '(item.barcode= ' + itemBarcode2 + ' and itemId = ' + extItemId2 + ' )'
+    When method GET
+    Then status 200
+    And match $.totalRecords == 1
+    And match $.requests[0].status == 'Closed - Cancelled'
+
+  #Negative
+  Scenario: Cancel DCB Transaction manually after ITEM_CHECKED_OUT
+    * def transactionId = '040'
+    * def itemId = 'c7a2f4de-77af-11ee-b962-0242ac120002'
+    * def createTransaction = call read('classpath:volaris/mod-dcb/reusable/create-dcb-transaction.feature') { transactionId: '#(transactionId)', extItemId: '#(extItemId3)', itemBarcode: '#(itemBarcode3)' }
+    * def updateToCancelRequest = read('classpath:volaris/mod-dcb/features/samples/transaction/update-dcb-transaction-to-cancel.json')
+
+    * def intCheckInDate = call read('classpath:volaris/mod-dcb/features/util/get-time-now-function.js')
+    * def checkInRequest = read('classpath:volaris/mod-dcb/features/samples/check-in/check-in-by-barcode-entity-request.json')
+    * set checkInRequest.itemBarcode = itemBarcode3
+
+
+    Given path 'circulation', 'check-in-by-barcode'
+    And request checkInRequest
+    When method POST
+    Then status 200
+    And match $.item.barcode == itemBarcode3
+    And match $.item.status.name == 'In transit'
+
+    * def updateToAwaitingPickupRequest = read('classpath:volaris/mod-dcb/features/samples/transaction/update-dcb-transaction-to-awaiting-pickup.json')
+
+    Given path 'transactions' , transactionId , 'status'
+    And request updateToAwaitingPickupRequest
+    When method PUT
+    Then status 200
+
+    * def updateToCheckOutRequest = read('classpath:volaris/mod-dcb/features/samples/transaction/update-dcb-transaction-to-item-check-out.json')
+
+    Given path 'transactions' , transactionId , 'status'
+    And request updateToCheckOutRequest
+    When method PUT
+    Then status 200
+
+    Given path 'transactions' , transactionId , 'status'
+    When method GET
+    Then status 200
+    And match $.status == 'ITEM_CHECKED_OUT'
+    And match $.role == 'LENDER'
+
+    Given path 'transactions' , transactionId , 'status'
+    And request updateToCancelRequest
+    When method PUT
+    Then status 400
+    And match $.errors[0].message == 'Cannot cancel transaction dcbTransactionId: 040. Transaction already in status: ITEM_CHECKED_OUT: '
+
+  Scenario: Cancel DCB Transaction manually after ITEM_CHECKED_IN
```

Comment:
I think after checked_in, cancellation should not be allowed

Reply:
fixed